### PR TITLE
Add detekt as kotlin linter, adjust code and fix some typos

### PR DESCRIPTION
**Pull request by ben-j69**
- add detekt to verify the formatting of the Kotlin code (used in local and in Travis CI)
- adjust some code from the detekt report : convert Utils and CodeGenerator to object
- fix typos in readme
- update tools version
- add .editorconfig for other contributors

This does not justify a new release but will prepare the project for next evolutions.

### CODE REVIEW

File: build.gradle.kts [hunk 1]
```diff
@@ -15,8 +16,8 @@ repositories {
 dependencies {
     implementation("com.android.tools.build:gradle:4.0.0")
 
-    testImplementation("io.kotlintest:kotlintest-runner-junit5:3.1.10")
-    testImplementation("junit:junit:4.13")
+    testImplementation("io.kotlintest:kotlintest-runner-junit5:3.1.11")
```
Comment (Mazorius) [inline build.gradle.kts:19]: I think kotlintest is renamed to kotest. Maybe this should be changed to kotest?

Comment (ben-j69) [reply build.gradle.kts:19]: Thanks but the rename is only for release 4.x, the project is not yet compatible for this version so I will let kotlintest in 3.1.11 until a real migration.
https://github.com/kotest/kotest/blob/master/doc/reference.md

File: README.md [hunk 2]
```diff
@@ -126,24 +126,27 @@ Secrets().getYourSecretKeyName(packageName)
...
 ```shell
-gradle copyCpp
-gradle copyKotlin [-Ppackage=your.package.name]
+./gradlew copyCpp
+./gradlew copyKotlin [-Ppackage=your.package.name]
 ```
 
 ### Obfuscate
 Create an obfuscated key and display it :
 ```shell
```
Comment (Mazorius) [inline README.md:135]: Gradle is supported by syntax engine.

```suggestion
``` gradle
```

File: README.md [hunk 3]
```diff
@@ -55,7 +55,7 @@ For more details about the installation check the [plugin's page](https://plugin
 
 Obfuscate and hide your key in your project :
 ```shell
```
Comment (Mazorius) [inline README.md:57]: Gradle is supported by syntax engine.

```suggestion
``` gradle
```

File: README.md [hunk 4]
```diff
@@ -94,7 +94,7 @@ As an example, we will use a [rot13 algorithm](https://en.wikipedia.org/wiki/ROT
 After a rot13 encoding your key `yourKeyToObfuscate` becomes `lbheXrlGbBoshfpngr`.
 Add it in your app :
 ```shell
```
Comment (Mazorius) [inline README.md:96]: Gradle is supported by syntax engine.

```suggestion
``` gradle
```

File: README.md [hunk 5]
```diff
@@ -126,24 +126,27 @@ Secrets().getYourSecretKeyName(packageName)
 ### Copy files
 Copy required files to your project :
 ```shell
```
Comment (Mazorius) [inline README.md:128]: Gradle is supported by syntax engine.

```suggestion
``` gradle
```

**Review by Mazorius — CHANGES_REQUESTED** [review]

**Comment by Mazorius** [comment]
The possible keywords are listet here: https://github.com/github/linguist/blob/master/lib/linguist/languages.yml

File: README.md (hunk 2 above)
Comment (ben-j69) [reply README.md:135]: @Mazorius I used `gradle` in the beginning of the readme for gradle files. This is a command line what will be the difference by annotating it as a gradle despite it is a shell/console command ?

Comment (Mazorius) [reply README.md:135]: Sorry - was my bad. gradle is for Gradle files and not shell commands.
You can ignore my suggestions 🤦

**Review by GhassenMsd — APPROVED** [review]

**Review by Mazorius — APPROVED** [review]

**Pull request merged**